tests/utils/exec.ts

```diff
@@ -410,42 +343,79 @@ async function pkExecTarget(
 }
 
 /**
- * This will spawn a process that executes the target `cmd` provided.
- * @param args - args to be passed to the command.
- * @param env - environment variables to be passed to the command.
- * @param cwd - the working directory the command will be executed in.
- * @param logger
+ * Launch pk command through subprocess
+ * This is the default
  */
-async function pkSpawnTarget(
+async function pkSpawnWithoutShell(
   args: Array<string> = [],
-  env: Record<string, string | undefined> = {},
-  cwd?: string,
-  logger: Logger = new Logger(pkSpawn.name),
+  opts: ExecOpts = { env: {} },
+  logger: Logger = new Logger(pkSpawnWithoutShell.name),
 ): Promise<ChildProcess> {
-  cwd = path.resolve(
-    cwd ??
+  const cwd =
+    opts.cwd ??
+    (await fs.promises.mkdtemp(path.join(globalThis.tmpDir, 'polykey-test-')));
+  const env = {
+    ...process.env,
+    ...opts.env,
+  };
+  // Recall that we attempt to connect to all specified seed nodes on agent start.
+  // Therefore, for testing purposes only, we default the seed nodes as empty
+  // (if not defined in the env) to ensure no attempted connections. A regular
+  // PolykeyAgent is expected to initially connect to the mainnet seed nodes
+  env['PK_SEED_NODES'] = env['PK_SEED_NODES'] ?? '';
+  const subprocess = childProcess.spawn(
+    'ts-node',
+    ['--project', tsConfigPath, polykeyPath, ...args],
+    {
+      env,
+      cwd,
+      stdio: ['pipe', 'pipe', 'pipe'],
+      windowsHide: true,
+      shell: opts.shell ? opts.shell : false,
+    },
+  );
+  // The readline library will trim newlines
+  const rlOut = readline.createInterface(subprocess.stdout!);
+  rlOut.on('line', (l) => logger.info(l));
+  const rlErr = readline.createInterface(subprocess.stderr!);
+  rlErr.on('line', (l) => logger.info(l));
+  return subprocess;
+}
+
+/**
+ * Launch pk command through subprocess
+ * This is the parameter > environment override
+ */
+async function pkSpawnWithShell(
+  args: Array<string> = [],
+  opts: ExecOpts = { env: {}, command: globalThis.testCmd },
+  logger: Logger = new Logger(pkSpawnWithShell.name),
+): Promise<ChildProcess> {
+  const cwd = path.resolve(
+    opts.cwd ??
       (await fs.promises.mkdtemp(
         path.join(globalThis.tmpDir, 'polykey-test-'),
       )),
   );
-  env = {
+  const env = {
     ...process.env,
-    ...env,
-    DOCKER_OPTIONS: generateDockerArgs(cwd).join(' '),
+    ...opts.env,
   };
+  if (globalThis.testPlatform === 'docker') {
+    env.DOCKER_OPTIONS = generateDockerArgs(cwd).join(' ');
+  }
```

Comment:
I'm wondering should this be part of the utilities, or should this be in the test cases? This seems weird to put here. It seems the test cases themselves should be the ones changing their execution options by checking the `testPlatform`. @tegefaulkes

Reply:
@tegefaulkes perhaps this change will need to be assigned into #435. Because at that point we can group all docker tests into `tests/integration/docker`. And there is where we should be doing things like the above. In fact... I believe the `PK_TEST_PLATFORM` would be removed too.

Reply:
I'm resistant to putting it in the test cases. It means adding it to a lot of places, having to update all those places if it changes and the worst aspect, having to remember to include it when writing new tests. Basically it would add more surface area for making mistakes.

I think it makes sense to include it along side other logic for switching behaviour based on when `globalThis.testCmd` is set.

Reply:
I've added some details into #435.

But now that I see this was added in this PR.

So can these things be in the test cases in `tests/bin`, or is there some other reason this is here? Perhaps to avoid having to write it out in each test case?

@emmacasolin was this piece of code brought in from a different location?

Reply:
Oh ok I see it now under `generateDockerArgs`. I reckon this is indeed something that would have to go into #435.